queue-worker/main.go

```diff
@@ -215,12 +223,14 @@ func main() {
 	var sub stan.Subscription
 
 	reconnectHandler = func(nc *nats.Conn) {
-		nc.Close()
-		sc, err = stan.Connect(clusterID, clientID, stan.NatsURL("nats://"+natsAddress+":4222"))
+		sc.Close()
+		nc.SetReconnectHandler(reconnectHandler)
```

Comment:
Remove these 2 lines (set reconnect and natsConn=nc). The reconnect handler's `nc *nats.Conn` will be the pointer to the connection you created earlier. No need to do anything with that.